src/Microsoft.IdentityModel.JsonWebTokens/JsonWebToken.cs

```diff
@@ -579,14 +582,41 @@ private void ReadToken(string encodedJson)
             EncodedToken = encodedJson;
         }
 
+        /// <inheritdoc/>
+        public override string SecureToString()
+        {
+            var sb = new StringBuilder();
+            if (!IsEncrypted)
+            {
+                sb.Append(EncodedHeader);
+                sb.Append('.');
+                sb.Append(EncodedPayload);
+            }
+            else
+            {
+                sb.Append(EncodedHeader);
+                sb.Append('.');
+                sb.Append(EncryptedKey);
+                sb.Append('.');
+                sb.Append(InitializationVector);
+                sb.Append('.');
```

Comment:
There are 4 dots in a JWT? I thought there were only 3?

Reply:
There are 5 segments for JWE and 3 segments for a JWS.